crates/turborepo-lib/src/task_graph/visitor.rs

```diff
@@ -881,6 +924,18 @@ impl ExecContext {
             }
         }
 
+        if !self.task_cache.is_caching_disabled() {
+            let missing_platform_env = self.platform_env.validate(&self.execution_env);
+            if !missing_platform_env.is_empty() {
+                let _ = self.warnings.lock().map(|mut warnings| {
+                    warnings.push(TaskWarning {
+                        task_id: self.task_id_for_display.clone(),
+                        missing_platform_env,
+                    });
+                });
```

Comment:
It's probably worth panicking here if locking fails. Locking should only fail if another thread panicked while holding the lock.
```suggestion
                self.warnings.lock().expect("warnings lock poisoned").push(TaskWarning {
                        task_id: self.task_id_for_display.clone(),
                        missing_platform_env,
                 });|
```